### ISSUE TITLE
Parent component state affects StaticImage fade animation

### ISSUE THREAD

**Issue by steverandy** [gatsbyjs/gatsby#32492]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

If parent component has a state change during mounting, the StaticImage will show fade animation.
However if the state is removed, StaticImage will not show any animation.
What is the correct behavior? Based on past experience using gatsby image component, it never show fade animation.

https://user-images.githubusercontent.com/525332/126751272-73895f09-23e1-4e13-b706-ea41467776a8.mov

### Reproduction Link

https://github.com/steverandy/gatsby-demo-image-issue

### Steps to Reproduce

1. npm run develop
2. Open page in Chrome browser
3. Click "Go to page 2"
4. Press browser back button

### Expected Result

After step 4, image should be shown without fading animation.

### Actual Result

After step 4, image is shown with fading animation.

### Environment

```shell
System:
    OS: macOS 12.0
    CPU: (6) x64 Intel(R) Core(TM) i5-8500 CPU @ 3.00GHz
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.3.0 - /usr/local/bin/node
    npm: 7.18.1 - /usr/local/bin/npm
  Languages:
    Python: 2.7.17 - /usr/local/bin/python
  Browsers:
    Chrome: 92.0.4515.107
    Firefox: 90.0
    Safari: 15.0
  npmPackages:
    gatsby: ^3.10.1 => 3.10.1
    gatsby-plugin-gatsby-cloud: ^2.10.0 => 2.10.0
    gatsby-plugin-image: ^1.10.1 => 1.10.1
    gatsby-plugin-manifest: ^3.10.0 => 3.10.0
    gatsby-plugin-offline: ^4.10.0 => 4.10.0
    gatsby-plugin-react-helmet: ^4.10.0 => 4.10.0
    gatsby-plugin-sharp: ^3.10.2 => 3.10.2
    gatsby-source-filesystem: ^3.10.0 => 3.10.0
    gatsby-transformer-sharp: ^3.10.0 => 3.10.0
  npmGlobalPackages:
    gatsby-cli: 3.10.0
```
